producer/writer/message_writer.go

```diff
@@ -218,13 +230,12 @@ func (w *messageWriterImpl) write(
 		break
 	}
 	m.DecReads()
-
-	if !written {
-		// Could not be written to any consumer, will retry later.
-		w.m.allConsumersWriteError.Inc(1)
-		return
+	if written {
+		return nil
 	}
-	m.SetRetryAtNanos(w.nextRetryNanos(m.WriteTimes(), nowNanos))
+	// Could not be written to any consumer, will retry later.
```

Comment:
What happened with retry logic here? Is that superseded with the valid connection logic?

Reply:
the retry is basically waiting for the next tick in the background go routine to go through the whole queue again.